src/main/java/hudson/plugins/sshslaves/JavaProvider.java

```diff
@@ -62,4 +71,35 @@
         return ExtensionList.lookup(JavaProvider.class);
     }
 
+    /**
+     * Gets minimal required Java version.
+     * 
+     * @return Minimal Java version required on the master and agent side.
+     *         It will be {@link #JAVA_LEVEL_7} if the core version cannot be determined due to whatever reason.
+     * @since TODO
+     * 
+     */
+    @Nonnull
+    public static VersionNumber getMinJavaLevel() {
```

Comment:
Except for the case of plugins that require newer Java than core, and are running on the master.